docs/docs/graphql-api.md

```diff
@@ -0,0 +1,242 @@
+
+**Note**: GraphQL isn't necessary, you can still [use Gatsby without GraphQL](/docs/using-gatsby-without-graphql/)
+
+Sourcing data with an existing plugin requires installing necessary packages and adding the plugin to the plugins array in the `gatsby-config` with any necessary configurations. To [source data from the filesystem](/docs/sourcing-from-the-filesystem/):
+
+```shell
+npm install --save gatsby-source-filesystem
+```
+
+```js:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: `Your Site Name`,
+  },
+  plugins: [
+    // highlight-start
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `src`,
+        path: `${__dirname}/src/`,
+      },
+    },
+    // highlight-end
+  ],
+}
+```
+
+You can also [create custom plugins](/docs/creating-plugins/) to fit your own use cases and pull in data however you want.
+
+## Querying for Data
+
+Data can be queried inside pages, components, or the `gatsby-node.js` file, using one of these options:
+
+- `pageQuery`
+- `StaticQuery`
+- `useStaticQuery`
+
+**Note**: Because of how Gatsby processes GraphQL queries, you can't mix page queries and static queries in the same file. You also can't have multiple page queries or static queries in one file.
+
+### `pageQuery`
+
+You can have one page query per page. A page query can take GraphQL arguments for variables in your queries. A [page is made in Gatsby](/docs/page-creation/) by any React component in the `src/pages` folder, or by calling the `createPage` action and using a component in the `createPage` options -- meaning a `pageQuery` won't work in any component, only components that meet this criteria.
+
+Also, refer to the [guide on querying data in pages with page query](/docs/page-query/).
+
+#### Params
+
+A page query isn't a method, but is assigned a `graphql` string with a valid query inside it as an exported constant:
+
+```javascript
+export const pageQuery = graphql`
+  query HomePageQuery {
+    site {
+      siteMetadata {
+        description
+      }
+    }
+  }
+`
+```
+
+**Note**: the `const` that gets exported doesn't need to be called `pageQuery`, it can have anything, Gatsby just looks for an exported `graphql` string from the file.
+
+#### Returns
+
+A page query returns a data object passed into the props of the page component of the file the query is in.
+
+```javascript
+// highlight-start
+const HomePage = ({ data }) => {
+  // highlight-end
+  return (
+    <div>
+      Hello!
+      {data.site.siteMetadata.description} // highlight-line
+    </div>
+  )
+}
+```
+
+### `StaticQuery`
+
+Gatsby provides a `StaticQuery` component that can be used in any component to get data from GraphQL. You can only have one `StaticQuery` per page, in order to include the data you need from multiple sources you can use one query with multiple root fields. It cannot take variables as arguments.
```

Comment:
```suggestion
StaticQuery is a built-in component for retrieving data from Gatsby’s data layer in non-page components, such as a header, navigation, or any other child component.

You can only have one `StaticQuery` per page: in order to include the data you need from multiple sources, you can use one query with multiple [root fields](). It cannot take variables as arguments.
```
Is there something we can link to about root fields? or add an explanation?

Reply:
I like [this blog post from Apollo](https://blog.apollographql.com/the-anatomy-of-a-graphql-query-6dffa9e9e747) that explains a bunch of those concepts. I don't see anything we have that explains root fields, maybe I add it to the GraphQL concepts page with some other explanations sort of like that blog post includes?

Reply:
I added this image and an explanation in the GraphQL concepts page that this can link to for now:

![image](https://user-images.githubusercontent.com/21114044/65079757-17715500-d95d-11e9-9583-5341b569c965.png)

I also added the above blog post to the Further Reading section since that's what I referenced in making a Gatsby specific, simpler example.